src/app/auth/error.interceptor.ts

```diff
@@ -0,0 +1,19 @@
+import { HttpErrorResponse, HttpInterceptorFn } from '@angular/common/http';
+import { catchError, throwError } from 'rxjs';
+
+export const errorInterceptor: HttpInterceptorFn = (req, next) => {
+  return next(req).pipe(
+    catchError((error: unknown) => {
+      if (error instanceof HttpErrorResponse) {
+        if (error.status === 401) {
+          console.error('Unauthorized request:', error);
```

Comment:
You will probably have to implement refreshing of tokens here. You can do that later